filebeat/prospector/log/stdin.go

```diff
@@ -2,17 +2,28 @@ package log
 
 import (
 	"os"
-
-	"github.com/elastic/beats/filebeat/harvester/source"
 )
 
 // Stdin reads all incoming traffic from stdin and sends it directly to the output
 
 func (h *Harvester) openStdin() error {
-	h.file = source.Pipe{File: os.Stdin}
+	h.source = Pipe{File: os.Stdin}
 
 	var err error
-	h.encoding, err = h.encodingFactory(h.file)
+	h.encoding, err = h.encodingFactory(h.source)
 
 	return err
 }
+
+// restrict file to minimal interface of FileSource to prevent possible casts
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported type Pipe should be of the form "Pipe ..." (with optional leading article)